microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/MicroProfileConfigHoverParticipant.java

```diff
@@ -101,57 +112,133 @@ public Hover collectHover(JavaHoverContext context, IProgressMonitor monitor) th
 			return null;
 		}
 
-		String propertyValue = JDTMicroProfileProjectManager.getInstance().getJDTMicroProfileProject(javaProject)
-				.getProperty(propertyKey, null);
+		JDTMicroProfileProject mpProject = JDTMicroProfileProjectManager.getInstance()
+				.getJDTMicroProfileProject(javaProject);
+		MicroProfileConfigPropertyInformation propertyInformation = getKeyValueMapWithProfiles(propertyKey, annotation,
+				mpProject);
+		return new Hover(getDocumentation(propertyInformation, context.getDocumentFormat(),
+				context.isSurroundEqualsWithSpaces()), propertyKeyRange);
+	}
+
+	/**
+	 * Returns a map with property keys and values, where keys include
+	 * <code>propertyKey</code> and its value. The map also includes
+	 * <code>propertyKey</code> with profiles and its respective values.
+	 *
+	 * For example if <code>propertyKey</code> is "greeting.message", here is a
+	 * possible resulting map:
+	 *
+	 * "greeting.message" -> "hello" "%dev.greeting.message"-> "hi"
+	 * "%prod.greeting.message"-> "good morning"
+	 *
+	 * @param propertyKey the property key
+	 * @param annotation  the annotation
+	 * @param project     the project
+	 * @return
+	 * @throws JavaModelException
+	 */
+	public static MicroProfileConfigPropertyInformation getKeyValueMapWithProfiles(String propertyKey, IAnnotation annotation,

```

Comment:
I would rename `IAnnotation annotation` with `IAnnotation configPropertyAnnotation` to b emore understandable